### ISSUE TITLE
[DOCS] Getting Started instructions are broken (Bad rustc version)

### ISSUE THREAD

**Issue by yeswalrus** [facebook/buck2#426]
Attempting to run the commands on the `Getting Started` page:
```
rustup install nightly-2023-05-28
cargo +nightly-2023-05-28 install --git https://github.com/facebook/buck2.git buck2
```
I see the following error:

```
error: failed to run custom build command for `buck2_core v0.1.0 (/home/wagray/.cargo/git/checkouts/buck2-881d6af740402932/5c69d8e/app/buck2_core)`

Caused by:
  process didn't exit successfully: `/tmp/cargo-installh0BlOm/release/build/buck2_core-9383ed164b222151/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'buck2 only works with version `rustc 1.72.0-nightly (36fb58e43 2023-06-26)` of rustc, but you are using `rustc 1.71.0-nightly (cca7ee581 2023-05-27)`. Correct version is installed automatically when `rustup` is used', app/buck2_core/build.rs:43:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  ```
  
  Worse, attempting to install using 2023-06-26 fails, because the latest update on that channel is `rust version 1.72.0-nightly (8084f397c 2023-06-25)`, which still triggers the failure

**Comment by JimCampagno** [comment]
@yeswalrus For the time being, this is how I got this to work. 

Source: https://github.com/facebook/buck2/blob/main/HACKING.md

```swift
git clone https://github.com/facebook/buck2.git
cd buck2/
cargo install --path=app/buck2
```

**Comment by KGrewal1** [comment]
Try cargo install the nightly day after the day that appears when compilation fails (2023-06-27). Had this issue yesterday, seems to install the nightly of the day before (so has installed 2023-06-25 for you)

**Comment by ndmitchell** [comment]
nightly-2023-07-10 is the right answer and what we have in our https://github.com/facebook/buck2/blob/main/rust-toolchain file. Alas, you have to have that on the command line to for Cargo to get it right. Let me update the docs.

**Comment by ndmitchell** [comment]
Fixed in 4e5e76998a3cb0636e415b047798dfdc1ac56c05 and docs regenerated. I've also got a patch pending to write down in rust-toolchain what needs to be done, to encourage it not to be forgotten next time.
